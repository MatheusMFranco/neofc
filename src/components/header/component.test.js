import React from 'react';
import { render } from 'react-native-testing-library';
import Header from '.';
import { fireEvent } from '@testing-library/react-native';

const mockedNavigation = jest.fn();
const mockedGoBack = jest.fn();

jest.mock('@react-navigation/native', () => {
  return {
    useNavigation: () => ({
    navigate: mockedNavigation,
      goBack: mockedGoBack,
    }),
  };
});

describe('Header', () => {
    beforeEach(() => {
        mockedNavigation.mockClear();
    });

    it('should render title', () => {
        const { getByText } = render(<Header title="Home" />);
        expect(getByText('Home')).toBeDefined();
    });

    it('should render back button component', () => {
        const { getByTestId } = render(<Header title="Home" back />);
        expect(getByTestId('back-button')).toBeDefined();
    });

    it('should call goBack function', () => {
        const { getByTestId } = render(<Header title="Home" back />);
        const button = getByTestId('back-button');
        fireEvent.press(button);
        expect(mockedGoBack).toHaveBeenCalledTimes(1);
    });

    it('should render config button', () => {
        const { getByTestId } = render(<Header title="Home" config />);
        expect(getByTestId('config-button')).toBeDefined();
    });

    it('should call config function', () => {
        const { getByTestId } = render(<Header title="Home" config />);
        const button = getByTestId('config-button');
        fireEvent.press(button);
        expect(mockedNavigation).toHaveBeenCalledTimes(1);
    });

    it('should render without buttons', () => {
        const { queryByTestId } = render(<Header title="Home" />);
        const backButton = queryByTestId('back-button');
        const configButton = queryByTestId('config-button');
        expect(backButton).toBeNull();
        expect(configButton).toBeNull();
    });

    test('should match the snapshot', () => {
        const tree = render(<Header title="Test Header" />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
