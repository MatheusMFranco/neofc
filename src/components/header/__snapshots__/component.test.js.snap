// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "rgba(255, 255, 255, 1)",
      "flexDirection": "row",
      "justifyContent": "space-between",
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "rgba(255, 255, 255, 1)",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        [
          {
            "textAlign": "left",
          },
          {
            "color": "rgba(28, 27, 31, 1)",
            "fontFamily": "System",
            "fontWeight": "400",
            "letterSpacing": 0,
          },
          {
            "writingDirection": "ltr",
          },
          {
            "color": "rgba(33, 0, 93, 1)",
            "fontSize": 16,
            "fontWeight": "bold",
            "padding": 24,
            "textAlign": "center",
          },
        ]
      }
      testID="title"
    >
      Test Header
    </Text>
  </View>
</View>
`;
